### PR TITLE
EZP-30575: Fixed parsing date format in formatICU

### PR DESCRIPTION
**Pull request by GrabowskiM**
| Question      | Answer
| ------------- | ---
| Tickets       | https://jira.ez.no/browse/EZP-30575
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | yno
| Tests pass?   | yes
| Doc needed?   | yes
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

Changed regex to match only repeatable chars for formatICU

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

**Comment by GrabowskiM** [comment]
@ezsystems/documentation-team there should probably be addition to documentation to show how to use date formats (and in this particular case, how to escape strings), it's all in ICU documentation https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classSimpleDateFormat.html#details

File: src/bundle/Resources/public/js/scripts/admin.format.date.js [hunk 1]
```diff
@@ -97,10 +97,29 @@
...
+
+    moment.fn.formatICU = function(format) {
+        /*
+            Splitting format by ' character in order to cut strings between ' ' as these strings should be escaped
+            according to ICU documentation: https://unicode-org.github.io/icu-docs/apidoc/released/icu4c/classSimpleDateFormat.html#details
+            At the end changing them to [ ] brackets for Moment.js
+        */
+        const form = format
+            .split('\'')
+            .map((partialStr, key) => {
```
Comment (lucasOsti) [inline src/bundle/Resources/public/js/scripts/admin.format.date.js]: Maybe you can shorthand for map
`.map((partialStr, key) => key % 2 ? partialStr : replaceFormat(partialStr))`

**Comment by lserwatka** [comment]
Could you merge it up?

**Comment by GrabowskiM** [comment]
Merge up done

**Pull request merged**